test/unittests/stack_test.cpp

```diff
@@ -130,3 +131,94 @@ TEST(stack, clear_on_empty)
     Stack<char> stack;
     stack.clear();
 }
+
+TEST(operand_stack, construct)
+{
+    OperandStack stack{0};
+    EXPECT_THAT(stack.size(), 0);
```

Comment:
Explicitly using `Eq` matcher arguably is more readable (you could read it like a sentence "expect that stack size equals 0")
```suggestion
    EXPECT_THAT(stack.size(), Eq(0));
```
but I think it's ok without it.